File: frontend/src/features/usergroup/model/CreateUserGroupRequest.ts
import type { UserGroupDescription } from '@/features/usergroup/model/UserGroupDescription'
import type { UserGroupName } from '@/features/usergroup/model/UserGroupName'
import type { UserGroupSlug } from '@/features/usergroup/model/UserGroupSlug'

export type CreateUserGroupRequest = {
  slug: UserGroupSlug
  name: UserGroupName
  description: UserGroupDescription
}

File: backend/src/main/scala/domains/usergroup/http/request/CreateUserGroupRequest.scala
package domains.usergroup.http.request

import domains.usergroup.model.*

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreateUserGroupRequest(
  slug: UserGroupSlug,
  name: UserGroupName,
  description: UserGroupDescription
)

object CreateUserGroupRequest:
  given Encoder[CreateUserGroupRequest] = deriveEncoder[CreateUserGroupRequest]
  given Decoder[CreateUserGroupRequest] = deriveDecoder[CreateUserGroupRequest]
